sig-scalability/blogs/k8s-services-scalability-issues.md

```diff
@@ -0,0 +1,131 @@
+# Known Scalability Issues with Kubernetes Services
+
+_by Shyam JVS, Google Inc (with inputs from Brian Grant & Wojciech Tyczysnki)_
+
+**March 2018**
+
+This document serves as a catalog of issues we've known/discovered with kubernetes services as of today, focusing on their scalability/performance. The purpose of the document is to make the information common knowledge for the community, so we can work together towards improving it. Listing them below in no particular order.
+
+
+## Performance of iptables when large no. of service ports exist
+
+#### Issue
+
+Iptables can be slow in the presence of a large no. of services. As the no. of service ports increases, the KUBE-SERVICES chain gets longer and since it’s also evaluated very frequently that can bring down the performance. Also, it seems that manipulating a chain with lots of rules (e.g. KUBE-SERVICES) takes noticeably longer than others.
+
+#### Relevant issues/links
+
+- [#48938](https://github.com/kubernetes/kubernetes/issues/48938)
+
+#### Possible Solution(s)
+
+- Moving from iptables to IPVS may help here if that works stably. We have the IPVS alternative [implemented](https://github.com/kubernetes/kubernetes/pull/46580), but it still hasn’t gone to GA. Also we need to measure/confirm performance benefits from it.
+
+
+## Kube-proxy slow in updating iptables when large no. of backends exist
+
+#### Issue
+
+When we’re running services worth a total of 120k endpoints (backends) on our 5k-node scalability tests, we’re noticing that kube-proxy is timing out while trying to do iptables-restore due to failing to acquire lock over iptables.
```

Comment:
Did you verify that it's "timing-out" not crashing or anything like that?

Reply:
I've sent https://github.com/kubernetes/test-infra/pull/7545 to verify. Let's see what happens.

Reply:
> Did you verify that it's "timing-out" not crashing or anything like that?

Yes. We've now verified this is the case of "timing out" - https://github.com/kubernetes/kubernetes/issues/48107#issuecomment-398018717